client/my-sites/checkout/composite-checkout/components/jetpack-pro-redirect-modal/index.tsx

```diff
@@ -49,6 +50,24 @@ export default function JetpackProRedirectModal( { redirectTo, productSourceFrom
 
 	const isAgencyPartner = useSelector( isAgencyUser );
 
+	// Function to record the event when the modal is displayed.
+	// It is in a separate useEffect to avoid unecessary re-renders.
+	useEffect( () => {
+		const recordShowEvent = () => {
+			dispatch( recordTracksEvent( 'jetpack_dashboard_agency_checkout_redirect_modal_show' ) );
```

Comment:
I think we can directly use the dispatch instead of creating this method.

Reply:
Yes, good point, it was part of my attempt to match the previous code, but in this context it doesn't seem necessary